src/compile/model.ts

```diff
@@ -280,10 +280,10 @@ export abstract class Model {
     return assembleLegends(this);
   }
 
-  public assembleGroup(signals: VgSignal[] = []) {
+  public assembleGroup(mixins: {signals?: VgSignal[], scales: VgScale[]}) {
```

Comment:
Perhaps worth adding a comment about why scales and signals are passed in (rather than assembled within this method as before)? Took me a second to understand why the change in logic was necessary (to allow for scale references from signals, which the previous approach did not).